square/builder.go

```diff
@@ -96,7 +96,13 @@ func (b *Builder) AppendBlobTx(blobTx *blob.BlobTx) bool {
 	// create a new blob element for each blob and track the worst-case share count
 	blobElements := make([]*Element, len(blobTx.Blobs))
 	maxBlobShareCount := 0
-	for idx, blob := range blobTx.Blobs {
+	for idx, protoBlob := range blobTx.Blobs {
+		blob, err := blob.NewFromProto(protoBlob)
+		if err != nil {
+			// TODO: we should look at having a go native BlobTx type
+			// that we have already verified instead of doing it twice here
+			panic(fmt.Sprintf("invalid blob %d: %v", idx, err))
```

Comment:
[not blocking nit]

can we add add a comment to this issue to indicate that it requires that all blobs passed to it are valid? 

either that or add a new method / break this api that allows for error handling. my personal preference is that we panic as high in the stack as possible as that makes the problem that much clearer.